plugins.json

```diff
@@ -15723,6 +15723,99 @@
+    "download_url": "https://plugins-store.test.mattermost.com/release/mattermost-plugin-azure-devops-v3.2.0.tar.gz",
+    "release_notes_url": "https://github.com/mattermost/mattermost-plugin-azure-devops/releases/3.2.0",
+    "hosting": "",
+    "author_type": "mattermost",
+    "release_stage": "production",
+    "enterprise": false,
+    "signature": "iQIzBAABCAAdFiEExViBuA9p6GO4WtXR0bVLR6XO/sQFAmQ4KHAACgkQ0bVLR6XO/sQx+RAAnfhp/Uymih07gn6deCpicSPLDkGSniXuMnsQpgct2QrGam8STa6++4/B45YmKrLyttuB+rv4stwqv1nXWemICCRowMAdg3HX8+hRmtzCKW0dTb4itY/HAOXDWfVifGTqFcJbRuwZRmsn/i8BZCH0u6yFzsvpUlKGyvcdkMWIWxW2u23076oDH5m+EZOwL4K3aawM2DeRw8TqbQlwHKxHcyUWNmmbrsjI5j4s/PaMs7N0m51sIz+cVRT8XwC+GUydDSIgVI/6bQ5R8fFab4TiteFRRo5f01uYeRfeqIHDYWWcIU0PmWwXU0696boxRqmLKHQohpt/f57w9UQK1+zvXPsO6Ugk1SrhgBqjgAfaWQFMxsa4Qr2hv8iGlBJz13jr0zi5yxGnEHoxikBWD6A+3J55EZw4VIuiq7Hc/neTnmIkvaf50VSb9XLwZiza1lUXC26J9tTTg0JCmCyn1ADj0rOgkXHp+U29TxrDhiTfcNXwWksIKrrSGj8KgWSeyvCRuHnOKgNk6h4Lkez8R+VkpmwXcx489C0ZkLEH7uVpAjMXOaZNmc1CCmlpyAaNVrMkG6CDJpzMFchCwzIyrSnVD3hP0GtVn03DLWlXCUAJLVMQsrZaJejF7hvFzI33ZsGrkLxKbcNG9eJipqtx1+zuv3mbU6/OvkDzkNGyA1h0LOo=",
+    "repo_name": "mattermost-plugin-azure-devops",
+    "manifest": {
+      "id": "mattermost-plugin-azure-devops",
+      "name": "Azure DevOps",
+      "description": "This plugin provides services of Azure DevOps such as Boards and Repos",
+      "homepage_url": "https://github.com/mattermost/mattermost-plugin-azure-devops",
+      "support_url": "https://github.com/mattermost/mattermost-plugin-azure-devops/issues",
+      "release_notes_url": "https://github.com/mattermost/mattermost-plugin-azure-devops/releases/3.2.0",
+      "icon_path": "public/assets/azurebot.svg",
+      "version": "3.2.0",
+      "min_server_version": "5.37.3",
+      "server": {
+        "executables": {
+          "darwin-amd64": "server/dist/plugin-darwin-amd64",
+          "darwin-arm64": "server/dist/plugin-darwin-arm64",
+          "linux-amd64": "server/dist/plugin-linux-amd64",
+          "linux-arm64": "server/dist/plugin-linux-arm64",
+          "windows-amd64": "server/dist/plugin-windows-amd64.exe"
```

Comment:
not familiar with this but, is it expected to have the 5 binaries here but only 3 in the platforms section?

Reply:
@trilopin These 5 executables are what the plugin was compiled to support. In the main bundle of the plugin's release, the `tar.gz` file contains all server executables listed here.

The `platforms` struct you're mentioning is an optimization that is currently out of sync with all the architectures we support. There's an open PR to synchronize these here https://github.com/mattermost/mattermost-marketplace/pull/238. During the execution of the `/mb cutplugin` command, Matterbuild [extracts out each executable individually](https://github.com/mattermost/matterbuild/blob/59f5f2df3571c981871ed21cf78a420d1934e539/server/plugin_release.go#L404-L408), and uploads these more bare bundles to S3. The URLs to the CDN serving those bundles are then used in the `platforms` struct.

This was originally done to improve Cloud server startup times, since syncing with the file store on startup was causing a lot of delay on startup. Since this feature was not made to help self-hosted servers, it hasn't been diligently maintained as such. Older servers that don't know about the optimization, and servers that are running on other architectures, are unaffected since it is optional and backwards-compatible.